### PR TITLE
support tx proposals with multiple outputs

### PR DESCRIPTION
**Pull request by greggzigler**

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -756,8 +766,22 @@ API.prototype.savePreferences = function(preferences, cb) {
...
+    var proposalHeader = {
+      outputs: _.map(args.outputs, function(output) {
+        return _.pick(output, ['toAddress', 'amount', 'message']);
+      }),
+      message: args.message,
+      payProUrl: args.payProUrl
+    };
+    hash = WalletUtils.getProposalHash(proposalHeader);
+  } else {
+    $.shouldBeNumber(args.amount);
```
Comment (isocolsky) [inline lib/api.js]: If this test is needed (I'm not so sure), it should also be performed on the `amount` attribute of the `outputs` array?

Comment (greggzigler) [inline lib/api.js]: good point, i'll add that test, to match the legacy test.

File: lib/api.js [hunk 2]
```diff
@@ -328,6 +332,12 @@ API.prototype._doRequest = function(method, url, args, cb) {
       depth: 10
     }));
 
+    if (!res) {
```
Comment (isocolsky) [inline lib/api.js]: Good catch!

File: test/client.js [hunk 3]
```diff
@@ -1076,6 +1090,56 @@ describe('client API', function() {
...
+          clients[0].sendTxProposal(opts, function(err, x) {
+            should.not.exist(err);
+            clients[0].getTx(x.id, function(err, x2) {
+              should.not.exist(err);
+              x2.creatorName.should.equal('creator');
+              x2.message.should.equal('hello');
+              x2.fee.should.equal(10000);
+              x2.outputs[0].toAddress.should.equal('n2TBMPzPECGUfcT2EByiTJ12TPZkhN2mN5');
+              x2.outputs[0].amount.should.equal(10000);
+              x2.outputs[0].message.should.equal('world');
```
Comment (isocolsky) [inline test/client.js]: Do you think it would improve the test if some checks are performed on the second output too?

Comment (greggzigler) [inline test/client.js]: sure, why not? i'll add that.

**Comment by isocolsky** [comment]
ACK. Manual testing ok.

**Pull request merged**